packages/flutter_web_plugins/lib/src/navigation/js_url_strategy.dart

```diff
@@ -42,7 +40,7 @@ typedef _HistoryMove = Future<void> Function(int count);
 
 /// Given a Dart implementation of URL strategy, converts it to a JavaScript
 /// URL strategy to be passed through JS interop.
-JsUrlStrategy convertToJsUrlStrategy(UrlStrategy strategy) {
+JsUrlStrategy? convertToJsUrlStrategy(UrlStrategy? strategy) {
```

Comment:
Unfortunately, the initial version of nullability cannot express "if the argument is not null, the return result is also not null" so everyone needs to deal with the nullable return even when it's impossible. Would it be possible to make this function take and return non-null values and push the null awareness to the call sites?

Reply:
That's a good idea. This function is only called in one place (and is not exposed outside of this package) so it should be straightforward to move the null check to the call site here: https://github.com/flutter/flutter/blob/ffcf41913655409e11c3cf2c28b08af906a5143e/packages/flutter_web_plugins/lib/src/navigation/url_strategy.dart#L18

Reply:
I like these conversion functions not having to handle the null case; this is going to be a very common pattern in the web plugins, that have *plenty* of these!